src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -296,5 +284,160 @@ internal SafeSslHandle(IntPtr validSslPointer, bool ownsHandle) : base(IntPtr.Ze
+                Interop.Crypto.BioSetShoudRetryReadFlag(bioHandle);
+            }
+
+            public void SetData(byte[] buffer, int offset, int length)
+            {
+                Debug.Assert(_bytesAvailable == 0);
+
+                _byteArray = buffer;
+                _offset = offset;
+                _bytesAvailable = length;
+            }
+
+            public int Read(Span<byte> output)
+            {
+                var bytesToCopy = Math.Min(output.Length, _bytesAvailable);
+                if (bytesToCopy == 0)
+                {
+                    return -1;
+                }
+
+                var span = new Span<byte>(_byteArray, _offset, bytesToCopy);
+                span.CopyTo(output);
+                _offset += bytesToCopy;
+                _bytesAvailable -= bytesToCopy;
+                return bytesToCopy;
+            }
+
+            //Bio is already released by the ssl object
+            private void Dispose(bool isDisposing)
+            {
+                if (_handle.IsAllocated)
+                {
+                    _handle.Free();
+                }
+            }
+
+            public void Dispose()
+            {
+                Dispose(true);
+                GC.SuppressFinalize(this);
+            }
+
+            ~ReadBioBuffer()
+            {
+                Dispose(false);
+            }
+        }
+
+        internal class WriteBioBuffer : IDisposable
+        {
+            private SafeBioHandle _bioHandle;
+            private GCHandle _handle;
+            private byte[] _byteArray;
+            private int _bytesWritten;
+            private bool _isHandshake;
+
+            internal WriteBioBuffer(SafeBioHandle bioHandle)
+            {
+                _bioHandle = bioHandle;
+                _handle = GCHandle.Alloc(this, GCHandleType.Normal);
+                Interop.Crypto.ManagedSslBio.BioSetGCHandle(_bioHandle, _handle);
+            }
+
+            public void SetData(byte[] buffer, bool isHandshake)
+            {
+                _byteArray = buffer;
+                _bytesWritten = 0;
+                _isHandshake = isHandshake;
+            }
+
+            public int TakeBytes(out byte[] output)
+            {
+                output = _byteArray;
+                return TakeBytes();
+            }
+
+            public int TakeBytes()
+            {
+                var bytes = _bytesWritten;
+                _bytesWritten = 0;
+                _byteArray = null;
+                return bytes;
+            }
+
+            public int Write(Span<byte> input)
+            {
+                //Only for the handshake do we dynamically allocate
+                //buffers for normal encrypt operations we use a fixed
+                //size buffer handed to us and loop to do all the needed
+                //writes. This should be changed for the handshake as well
+                //but will require more securechannel/sslstatus changes
```

Comment:
Spot checking, it looks like we generally have

```C#
// slash-slash-space for text comments
/// slash-slash-slash-space when in doc comments
//slash_slash_nospace = commented_out_code;
```

It's not in the style guide per se, and there are definitely examples of "no space" for comments; but it seems to be the prevailing style.  (And I find easier to read, I guess, since this stood out to me as "weird")